_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+```
+
+As instructed by the event message, I contacted DigitalOcean support and they fixed it.
+
+### Find your gateway's external IP and update redirect URIs
+
+Once all the pods are running, find the gateway external IP with the `kubectl describe` command:
+
+```bash
+kubectl describe service gateway -n demo
+```
+
+```text
+Name:                     gateway
+Namespace:                demo
+Labels:                   app=gateway
+Annotations:              kubernetes.digitalocean.com/load-balancer-id: e81d2b8c-6c28-430d-8ba8-6dab29a1ba76
+                          service.beta.kubernetes.io/do-loadbalancer-certificate-id: bd0b1d03-0f90-449d-abbe-ac6a4026c133
+Selector:                 app=gateway
+Type:                     LoadBalancer
+IP Family Policy:         SingleStack
+IP Families:              IPv4
+IP:                       10.245.47.51
+IPs:                      10.245.47.51
+LoadBalancer Ingress:     157.230.200.181
+Port:                     http  8080/TCP
+TargetPort:               8080/TCP
+NodePort:                 http  31048/TCP
+Endpoints:                10.244.1.42:8080
+Session Affinity:         None
+External Traffic Policy:  Cluster
+```
+
+Update the redirect URIs in Okta to allow the gateway address as a valid redirect. Run `okta login`, open the returned URL in your browser, and sign in to the Okta Admin Console. Go to the **Applications** section, find your application, and edit it.
+
+Navigate to `http://<load-balancer-ingress-ip>:8080`.
+
+### Inspect and manage cluster resources
+
+You can retrieve the cluster details with the following `doctl` commands:
+
+```bash
+doctl k cluster get do1
+```
+```
+ID                                      Name    Region    Version        Auto Upgrade    HA Control Plane    Status     Endpoint                                                               IPv4             Cluster Subnet    Service Subnet    Tags                                            Created At                       Updated At                       Node Pools
+f59fc483-4ee4-4b59-be2c-bb017a932941    do1     nyc1      1.22.8-do.1    false           false               running    https://f59fc483-4ee4-4b59-be2c-bb017a932941.k8s.ondigitalocean.com    137.184.48.14    10.244.0.0/16     10.245.0.0/16     k8s,k8s:f59fc483-4ee4-4b59-be2c-bb017a932941    2022-05-03 20:40:01 +0000 UTC    2022-05-07 00:01:41 +0000 UTC    do1-default-pool
+```
+```bash
+ doctl k cluster list-associated-resources do1
+```
+```
+Volumes                                                                                                                                                                                       Volume Snapshots    Load Balancers
+[a33dbaba-cb22-11ec-9383-0a58ac145375 a6e18809-cb22-11ec-8723-0a58ac14468c 65fb3778-cb23-11ec-9383-0a58ac145375 ac106c49-cb22-11ec-9383-0a58ac145375 c7122288-cb22-11ec-bd08-0a58ac14467d]    []                  [8ebbcbf2-1e67-46f5-b38a-eddae17f00f3]
+```
```

Comment:
Why are these two commands useful?

Reply:
`cluster get` is just info, `cluster list-associated-resources` shows resources that are not deleted with the cluster and generate charges. That is mentioned some lines ahead.

Reply:
Removed the first command, and added a comment about the second.